### PR TITLE
✅ add static analysis of helix config

### PR DESCRIPTION
**Pull request by filmaj**
This PR ensures we:

- maintain a particular naming strategy for strains; all strains need to be named after the environment they will appear in (either `staging` or `production`)
- double check URLs contain valid values (that they are mounted either on our staging or production domain)

Please review @stevengill.

FYI @hirenoble - these tests should now catch the previous error we got surprised by when we missed using proper URLs in the strain you added in your last PR.

FYI @trieloff - these tests should ensure we have consistent naming and URLs for all strains moving forward, which should enable what we discussed in the CI/CD issue https://github.com/adobe/helix-cli/issues/821

### CODE REVIEW

File: test/helix-config/test.config.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+const assert = require('assert');
+const fs = require('fs');
+const yaml = require('js-yaml');
+
+const config = yaml.safeLoad(fs.readFileSync('helix-config.yaml', 'utf8'));
```
Comment (trieloff) [inline test/helix-config/test.config.js]: I'd load this from `helix-shared`, just to be sure to have the same interpretation.

Comment (filmaj) [reply test/helix-config/test.config.js]: Hmm, I suppose that the runtime-tests we have under `tests/runtime` should also use helix-shared to load the yaml and not js-yaml... noted.

**Comment by filmaj** [comment]
@trieloff I must admit that I am not fluent with JSON schema though I do understand its value proposition. Perhaps that is a better mechanism for validation than these yaml-parsing JS tests? Not sure. Perhaps that's just a matter of taste? I'm open to it but have no opinion on it either way.

File: test/helix-config/test.config.js (hunk 1 above)
Comment (filmaj) [reply test/helix-config/test.config.js]: Not sure how to go about using `helix-shared`, filed https://github.com/adobe/helix-shared/issues/108 with my trials and tribulations

**Pull request merged**